### PR TITLE
Multilingual Wrapper for Zero-Shot-Retrieval

### PR DESCRIPTION
**Pull request by FreddeFrallan**
A simple wrapper to allow users to specify a translation mapping of the original captions, using a JSON file.

### CODE REVIEW

File: clip_benchmark/cli.py [hunk 1]
```diff
@@ -7,37 +7,51 @@
...
 
+
 def main():
     parser = argparse.ArgumentParser()
     parser.add_argument('--dataset', type=str, default="cifar10", help="Dataset to use for the benchmark")
     parser.add_argument('--split', type=str, default="test", help="Dataset split to use")
-    parser.add_argument('--model', type=str, default="ViT-B-32-quickgelu", help="Model architecture to use from OpenCLIP")
-    parser.add_argument('--pretrained', type=str, default="laion400m_e32", help="Model checkpoint name to use from OpenCLIP")
-    parser.add_argument('--task', type=str, default="zeroshot_classification", choices=["zeroshot_classification", "zeroshot_retrieval", "linear_probe"])
+    parser.add_argument('--model', type=str, default="ViT-B-32-quickgelu",
```
Comment (rom1504) [inline clip_benchmark/cli.py:19]: would be great if you could undo all these line end changes

File: clip_benchmark/datasets/multilingual_dataset.py [hunk 2]
```diff
@@ -0,0 +1,31 @@
+import json, os
+import torch
+
+class Multilingual_Dataset(torch.utils.data.Dataset):
```
Comment (rom1504) [inline clip_benchmark/datasets/multilingual_dataset.py:4]: downloading captions should be automated though, see builder.py with examples

File: clip_benchmark/cli.py (hunk 1 above)
Comment (FreddeFrallan) [reply clip_benchmark/cli.py:19]: Ahhh, I see my PyCharm environment is wrecking things up.

File: clip_benchmark/datasets/multilingual_dataset.py (hunk 2 above)
Comment (FreddeFrallan) [reply clip_benchmark/datasets/multilingual_dataset.py:6]: I've automated the download process to a large extent.
However, I find that there are 146 images in the multilingual coco dataset that don't seem to exist in the 2014 split... Could it be that the dataset incorporates captions+images from later splits?
[Original Github](https://github.com/adobe-research/Cross-lingual-Test-Dataset-XTD10)

**Comment by FreddeFrallan** [comment]
In addition to the MS-COCO dataset, I made the loading of classnames and classification templates optional. Meaning they are only loaded if we're doing a classification task.
This is to avoid a scenario where a multilingual retrieval dataset is loaded, that has no matching classnames/templates.

File: clip_benchmark/datasets/multilingual_dataset.py (hunk 2 above)
Comment (FreddeFrallan) [reply clip_benchmark/datasets/multilingual_dataset.py:6]: I've made it possible to still execute the evaluation without these, but this should obviously be investigated further.

**Pull request closed without merging**